tcl.html.markdown

```diff
@@ -0,0 +1,377 @@
+---
+language: Tcl
+contributors:
+    - ["Poor Yorick", "http://pooryorick.com/"]
+---
+
+Tcl was created by [John Ousterhout](http://wiki.tcl.tk/John Ousterout) as a
+reusable scripting language for chip design tools he was creating.  In 1997 he
+was awarded the [ACM Software System
+Award](http://en.wikipedia.org/wiki/ACM_Software_System_Award) for Tcl.  Tcl
+can be used both as an embeddable scripting language and as a general
+programming language.  It can also be used as a portable C library, even in
+cases where no scripting capability is needed, as it provides data structures
+such as dynamic strings, lists, and hash tables.  The C library also provides
+portable functionality for loading dynamic libraries, string formatting and
+code conversion, filesystem operations, network operations, and more.
+
+Tcl is a pleasure to program in.  Its discipline of exposing all programmatic
+functionality as commands, including things like loops and mathematical
+operations that are usually baked into the syntax of other languages, allows it
+to fade into the background of whatever domain-specific functionality a project
+needs.  Its design of exposing all values as strings, while internally caching
+a structured representation, bridges the world of scripting and systems
+programming in the best way.  Even Lisp is more syntactically heavy than Tcl.

```

Comment:
All values are strings.  Any string that conforms to list formatting requirements *is* a list. See http://wiki.tcl.tk/3018